client/app/components/QuerySelector.jsx

```diff
@@ -14,14 +14,14 @@ const { Option } = Select;
 function search(term) {
   // get recent
   if (!term) {
-    return Query.recent().$promise.then(results => {
+    return Query.recent().then(results => {
       const filteredResults = results.filter(item => !item.is_draft); // filter out draft
       return Promise.resolve(filteredResults);
     });
```

Comment:
In this and other places `return Promise.resolve(...)` was used because `$q` had some issues with promise chaining (ATM I don't remember which, but returning new promise solved them). Now, when `$q` is not used anymore, such code could be simplified:
```javascript
return Query.recent().then(results => results.filter(item => !item.is_draft)); // filter out draft
```